### ISSUE TITLE
[Security] Password disclosed in log

### ISSUE THREAD

**Issue by dfigus** [SukramJ/hahomematic#305]
Make sure you have read the documentation at https://github.com/danielperna84/custom_homematic#custom_homematic.

Make sure you are running the latest version of hahomematic or it's custom_component (and Home Assistant) before reporting an issue.

**custom_component/hahomematic version (if applicable):**
0.34.2

**Home Assistant version (if applicable):**
2022.2.6 (core in VENV)

**CCU version:**
- Model (CCU2/3, Raspi3/4, NUC, ProxmoxVE, ...): RPI3 B+
- OS (Original, RaspberryMatic, debmatic/...): Rapsbian 10 (Buster) + piVCCU3
- Version: Rapsbian 5.10.63-v7+ & piVCCU3 3.61.7

HA and piVCCU3 are on the same RPI

**Problem-relevant configuration:**

Do you use tls? No
Do you use callback? No
Do you use username and password? Yes
Which interfaces do you use? HmIP

**Describe the bug**
In the log I have one entry which discloses the password:
ERROR (MainThread) [hahomematic.client] ping: failed for ProxyException [(<ProtocolError for <user>:<secret_password>@192.168.178.42:2010/RPC2: 401 Unauthorized>,)]

Maybe there are other places where the password is written to the log

**To Reproduce**
Sorry, no idea. I just saw this in the log.

**Relevant HA log entries**
```
ERROR (MainThread) [hahomematic.client] ping: failed for ProxyException [(<ProtocolError for <user>:<secret_password>@192.168.178.42:2010/RPC2: 401 Unauthorized>,)]

```
Actual user and password masked... 

**Expected behavior**
No passwords are written to the log

**Screenshots**

**Additional context**

**Comment by dfigus** [comment]
I found a way to reproduce the issue. If the password is incorrect (e.g. change the password for the user on the CCU), then you will get the above mentioned 401 Unauthorized error.

**Comment by SukramJ** [comment]
Good catch.
I hope it is fixed in 0.35.2

**Comment by dfigus** [comment]
The ping errors are password free with 0.35.2. Changing the password on the CCU3 when HA was running results in the following log messages:
```
2022-02-23 11:14:32 ERROR (MainThread) [hahomematic.client] ping: failed for ProxyException [((<ProtocolError for ha:<PASSWORD>@192.168.178.42:2010/RPC2: 401 Unauthorized>,),)]
2022-02-23 11:14:32 ERROR (MainThread) [hahomematic.client] ping: failed for ProxyException [((<ProtocolError for ha:<PASSWORD>@192.168.178.42:9292/groups: 401 Unauthorized>,),)]
2022-02-23 11:14:32 ERROR (MainThread) [hahomematic.client] ping: failed for ProxyException [((<ProtocolError for ha:<PASSWORD>@192.168.178.42:2010/RPC2: 401 Unauthorized>,),)]
2022-02-23 11:14:32 ERROR (MainThread) [hahomematic.client] ping: failed for ProxyException [((<ProtocolError for ha:<PASSWORD>@192.168.178.42:9292/groups: 401 Unauthorized>,),)]
```
Looks good! Thanks!

I will keep an eye on the log messages and create a new issue in case I find further places.
